core/renderers/zelos/drawer.js

```diff
@@ -45,6 +45,13 @@ Blockly.zelos.Drawer = function(block, info) {
 Blockly.utils.object.inherits(Blockly.zelos.Drawer,
     Blockly.blockRendering.Drawer);
 
+/**
+ * @override
+ */
+Blockly.zelos.Drawer.prototype.draw = function() {
+  this.block_.pathObject.clearOutlines();
```

Comment:
Is there a way to avoid clearing and recreating these every time?  As operations go, creating and deleting DOM nodes seems expensive.

Reply:
There should be. The main problem is not knowing when an input has been deleted of the block. What we could do is mark that the path object at the start of the draw to watch what inputs get used, and remove any remaining ones at the end of the draw.